Classes/Wwwision/Neos/MailChimp/Domain/Service/MailChimpService.php

```diff
@@ -76,6 +83,44 @@ public function getListById($listId)
 
     /**
      * @param string $listId
+     * @param string $interestCategoryId
+     * @return array
+     */
+    public function getCategoryByListIdAndInterestCategoryId($listId, $interestCategoryId)
+    {
+        $cacheKey = "MailChimp_List_" . $listId ."_Category_" . $interestCategoryId;
```

Comment:
Besides we should probably add the vendor namespace (Wwwision) to the cache key to prevent any clashes. And lastly we should hash `listId` and `interestCategoryId` to prevent invalid cache identifiers.